runtime/v2/builder.go

```diff
@@ -228,3 +225,9 @@ func AppBuilderWithPostTxExec[T transaction.Tx](
 		a.postTxExec = postTxExec
 	}
 }
+
+func AppBuilderWithStoreOptions[T transaction.Tx](opts rootstore.Options) AppBuilderOption[T] {
```

Comment:
Thinking about it longer, maybe we could just do a _manual unmarshal_: basically setting the store options struct manually by getting stuff only.
I don't like the extra logic that is required in app.go personally actually.

Reply:
Sorry I'm not following your proposal, what does a _manual unmarshal_ look like here?

Reply:
For reference, I moved this code snippet (previously in `runtime/v2/builder.go`) to simapp/v2/app_di.go:
https://github.com/cosmos/cosmos-sdk/blob/5731e34a403a9d902e11f5529ca7cf4b3f9d4ca1/simapp/v2/app_di.go#L193-L198

When I tried to use `serverv2.UnmarshalSubConfig` the input struct was unmodified.  I didn't investigate why since the above snippet works.
https://github.com/cosmos/cosmos-sdk/blob/5731e34a403a9d902e11f5529ca7cf4b3f9d4ca1/server/v2/config.go#L42-L73

Reply:
Basically doing:

```go
func getStoreOptions(dynamicCfg server.dynamicCfg) rootstore.Options {
	storeOpts := rootstore.DefaultStoreOptions()
	if ssType := a.viper.GetString("store.options.sqlite"); ssType != "" {
		storeOpts.SSType = rootstore.SSType(ssType)
	}
	if scType := a.viper.GetString("store.options.iavl"); scType != "" {
		storeOpts.SCType = rootstore.SCType(scType)
	}

	// ...

	return storeOpts
}
```

directly in runtime. This way we don't need the user to add that in their app.go.

Reply:
May be overkilled, but I like the idea of not adding any manual work for the user in app.toml